ext/gd/tests/avif_decode_encode.phpt

```diff
@@ -0,0 +1,57 @@
+--TEST--
+avif decoding/encoding tests
+--SKIPIF--
+<?php
+    if (!extension_loaded('gd')) {
+        die("skip gd extension not available.");
+    }
+
+    if (!function_exists("imagecreatefrompng") || !function_exists("imagepng")) {
+        die("skip png support unavailable");
+    }
+    if (!function_exists("imagecreatefromavif") || !function_exists("imageavif")) {
+        die("skip avif support unavailable");
+    }
+?>
+--FILE--
+<?php
+    $cwd = __DIR__;
+    $infile = $cwd . '/girl.avif';
+    $outfile = $cwd . '/test.avif';
+
+    echo 'Decoding AVIF image: ';
+    $img = imagecreatefromavif($infile);
+    echo_status($img);
+
+    echo 'Default AVIF encoding: ';
+    echo_status(imageavif($img, $outfile));
+
+    echo 'Encoding AVIF at quality 70: ';
+    echo_status(imageavif($img, $outfile, 70));
+
+    echo 'Encoding AVIF at quality 70 with speed 5: ';
+    echo_status(imageavif($img, $outfile, 70, 5));
+
+    echo 'Encoding AVIF with default quality: ';
+    echo_status(imageavif($img, $outfile, -1));
+
+    echo 'Encoding AVIF with illegal quality: ';
+    echo_status(imageavif($img, $outfile, 1234));
+
+    echo 'Encoding AVIF with illegal speed: ';
+    echo_status(imageavif($img, $outfile, 70, 1234));
+
+    unlink($outfile);
+
+    function echo_status($success) {
+        echo $success ? "ok\n" : "failed\n";
+    }
+?>
```

Comment:
I don't think this test makes much sense, since it doe not checkl the actual image results. See e.g. https://github.com/php/php-src/blob/master/ext/gd/tests/webp_basic.phpt for a test that does this, still leaving room for minor variations due to lossy encoding.

Reply:
True, my tests just make sure that the AVIF functions exist, that they accept parameters we'd expect, that they handle a couple of instances of bad input, and that they return `true`. I did notice that there were other tests in the repo like the `webp_basic.phpt`, which actually ensure that an image can be encoded and then decoded without changing too much. I admit I didn't jump at the chance to write this test, since I had real trouble with subtle lossiness in libgd... finally I became pretty sure that this lossiness happened in libgd's PNG encoding/decoding, not what i was doing with AVIF :)

Regardless I'm happy to give this a try! Honestly, I didn't experience lossiness in libgd with basic images of the sort that webp test generates.

Reply:
good news - it passed. Now seeing if it can pass more stringent im↔im tests...

Reply:
Seems that 0 pixels changed. This image gave me that result in libgd too. So I wrote the test to expect that.

Let me know if you think it should be more lenient, if that test should go in its own file, or if I should remove any of the other tests, etc.

Reply:
That looks good to me. If there'll be some small differences in some cases, we can still easily adjust the test.